plugins/org.jboss.reddeer.workbench/src/org/jboss/reddeer/workbench/impl/editor/internal/BasicEditorFile.java

```diff
@@ -0,0 +1,75 @@
+/******************************************************************************* 
+ * Copyright (c) 2017 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/
+package org.jboss.reddeer.workbench.impl.editor.internal;
+
+import java.io.InputStream;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.runtime.CoreException;
+import org.jboss.reddeer.workbench.api.EditorFile;
+import org.jboss.reddeer.workbench.exception.WorkbenchLayerException;
+
+/**
+ * Basic implementation of EditorFile which represents a file associated with an
+ * editor.
+ * 
+ * @author Andrej Podhradsky (apodhrad@redhat.com)
+ *
+ */
+public class BasicEditorFile implements EditorFile {
+
+	private IFile iFile;
+
+	/**
+	 * Constructs the a implementation of EditorFile from IFile.
+	 * 
+	 * @param iFile
+	 *            IFile
+	 */
+	public BasicEditorFile(IFile iFile) {
+		this.iFile = iFile;
+	}
+
+	@Override
+	public InputStream getInputStream() {
+		InputStream inputStream = null;
+		try {
+			inputStream = iFile.getContents();
+		} catch (CoreException ce) {
+			throw new WorkbenchLayerException("Cannot get an input stream from the editor file", ce);
+		}
+		return inputStream;
```

Comment:
wouldn't be better throwing an exception that something went wrong instead of returning null? Also if null iFile is passed in constructor you will get NPE there.